src/core/load-fragment.c

```diff
@@ -6141,6 +6141,28 @@ int config_parse_mount_node(
         return config_parse_string(unit, filename, line, section, section_line, lvalue, ltype, path, data, userdata);
 }
 
+int config_parse_concurrency_max(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        unsigned *concurrency_max = ASSERT_PTR(data);
+
+        if (isempty(rvalue) || streq(rvalue, "infinity")) {
+                *concurrency_max = UINT_MAX;
+                return 0;
+        }
+
+        return config_parse_unsigned(unit, filename, line, section, section_line, lvalue, ltype, rvalue, data, userdata);
```

Comment:
Don't accept soft concurrency 0? To prevent DoSing job queue. (It shouldn't be real DoS but leg-shooting but better safe than sorry.) Or at least soft=0 && hard=infinity.

Reply:
i explicitly wanted to allow that, so that you can prep a slice that you can change later to unblock things...

Reply:
Don't you confuse .slice (~resources) and .target (~jobs) here?
I'm only curious what usage do you envision with this here?

Reply:
That pausing/unpausing sounds like freezing/unfreezing that's already there. (It also resembles [snapshots](https://manpages.debian.org/jessie/systemd/systemd.snapshot.5.en.html) RIP.)
All in all, accepting concurrency zero makes no harm.